### ISSUE TITLE
Client enumeration possible for client_credentials grant

### ISSUE THREAD

**Issue by hotab** [go-oauth2/oauth2#106]
If we do a request with non-existing client_id in the sample server, we get the following response:
```
$ curl 'http://localhost:9096/token?grant_type=client_credentials&client_id=000000000&client_secret=11111111&scope=all'
{"error":"server_error","error_description":"The authorization server encountered an unexpected condition that prevented it from fulfilling the request"}
```
On the other hand, if we do a request with existing client_id, but invalid client_secret, we get the following response:
```
$ curl 'http://localhost:9096/token?grant_type=client_credentials&client_id=000000&client_secret=111111&scope=all'
{"error":"invalid_client","error_description":"Client authentication failed"}
```

This can be used to enumerate valid client_id's, which might be undesirable.

**Comment by adeturner** [comment]
Agree, it is useful info for server side debugging though... and you can override the error handling to stop that returning to the client.

	s.srv.SetInternalErrorHandler(func(err error) (re *errors.Response) {
	s.srv.SetResponseErrorHandler(func(re *errors.Response) {

e.g. 

	s.srv.SetInternalErrorHandler(func(err error) (re *errors.Response) {
		re = &errors.Response{
			Error:       err,
			ErrorCode:   http.StatusInternalServerError,
			Description: "Hacking is a crime",
			StatusCode:  http.StatusInternalServerError,
		}
		return
	})

$ curl http://localhost:8080/token?grant_type=client_credentials\&client_id=${CLIENT_ID}\&client_secret=${CLIENT_SECRET}\&scope=read
{"error":"not found","error_code":500,"error_description":"Hacking is a crime"}
